internal/services/applicationinsights/application_insights_resource.go

```diff
@@ -371,7 +375,11 @@ func resourceApplicationInsightsRead(d *pluginsdk.ResourceData, meta interface{}
 		d.Set("force_customer_storage_for_profiler", props.ForceCustomerStorageForProfiler)
 
 		if v := props.WorkspaceResourceID; v != nil {
-			d.Set("workspace_id", v)
+			id, err := workspaces.ParseWorkspaceIDInsensitively(*v)
+			if err != nil {
+				return err
+			}
+			d.Set("workspace_id", id.ID())
 		}
```

Comment:
we'll want to ensure a value is always set into the state for this:

```suggestion
		workspaceId := ""
		if v := props.WorkspaceResourceID; v != nil {
			id, err := workspaces.ParseWorkspaceIDInsensitively(*v)
 			if err != nil {
 				return err
 			}
 			workspaceId = id.ID()
		}
		d.Set("workspace_id", workspaceId)
```